### PR TITLE
Add SAST with Octoscan

### LINKED ISSUE #33: Add static security scanning of GitHub Actions
Options:

* https://github.com/snyk-labs/github-actions-scanner
* https://github.com/synacktiv/octoscan
* https://github.com/woodruffw/zizmor
* https://github.com/boostsecurityio/poutine

### PR DESCRIPTION
**Pull request by johnbillion**
See #33

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -25,3 +25,34 @@ jobs:
...
+      run: | #shell
+        cd octoscan
+        go mod tidy
+        go build
+
+    - name: Run Octoscan
+      run: | #shell
+        # Octoscan does not use exit codes to indicate success or failure, so we need to check the output
+        ./octoscan/octoscan scan . | tee octoscan-result.txt
+        test ! -s octoscan-result.txt
```
Comment (szepeviktor) [inline .github/workflows/lint.yml]:
```suggestion
        ! ./octoscan/octoscan scan .github/workflows/ | grep '^'
```

Comment (johnbillion) [reply .github/workflows/lint.yml]: The path needs to point to the repo root, not the workflows dir. Otherwise looks good.

Comment (johnbillion) [reply .github/workflows/lint.yml]: @szepeviktor It looks like `'^'` is actually grepping for the start of a line, not the literal `^` character. Needs the `-F` flag I think.

Comment (johnbillion) [reply .github/workflows/lint.yml]: Going to leave this as-is 👍

Comment (szepeviktor) [reply .github/workflows/lint.yml]: `-F` causes grep to take `^` literally!!

Comment (szepeviktor) [reply .github/workflows/lint.yml]: carot is for the start of the line, meaning "anything"

Comment (johnbillion) [reply .github/workflows/lint.yml]: `test ! -s` does the same thing

Comment (szepeviktor) [reply .github/workflows/lint.yml:58]: The above test fails for a non-empty output.
If you need an empty output remove `!`.

Comment (szepeviktor) [reply .github/workflows/lint.yml]: All right.

**Comment by hugo-syn** [comment]
Hi, I've just released an action for `octoscan`, if you want to try it, it's here: https://github.com/synacktiv/action-octoscan

If you test it, I would appreciate your feedback to help me improve it :)

**Comment by johnbillion** [comment]
@hugo-syn Thanks! I'll take a look.

The main issue I'm facing with using Octoscan as part of CI is how to balance ignoring acceptable risks and identifying genuine issues. One example is expression injection from potentially untrusted output in another job. I'm hesitant to trim down the list of scans that happen in case it causes a genuine issue to go unreported.

**Comment by hugo-syn** [comment]
I understand, it was initially developed for offensive purpose so it's ok to get false positives. You can still use the `--ignore` option to filter specific errors.

**Pull request merged**